src/main/java/org/alliancegenome/curation_api/model/document/builders/ExpressionDetailDocumentBuilder.java

```diff
@@ -0,0 +1,51 @@
+package org.alliancegenome.curation_api.model.document.builders;
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import java.util.Map;
+
+import org.alliancegenome.curation_api.model.document.es.ExpressionDetailDocument;
+import org.alliancegenome.curation_api.model.entities.GeneExpressionAnnotation;
+import org.alliancegenome.curation_api.model.entities.GeneExpressionExperiment;
+import org.alliancegenome.curation_api.model.ingest.dto.fms.ConsolidatedGeneExpressionFmsDTO;
+import org.alliancegenome.curation_api.services.ReferenceService;
+import org.alliancegenome.curation_api.services.helpers.annotations.GeneExpressionAnnotationUniqueIdHelper;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import lombok.NoArgsConstructor;
+
+@NoArgsConstructor
+@RequestScoped
+public class ExpressionDetailDocumentBuilder {
+
+	@Inject private GeneExpressionAnnotationUniqueIdHelper geneExpressionAnnotationUniqueIdHelper;
+	@Inject private ReferenceService referenceService;
+
+	public ExpressionDetailDocument build(GeneExpressionAnnotation annotation, Map<String, GeneExpressionExperiment> experiments) {
```

Comment:
Seems to be a lot of potential for NPEs in this class with chained get calls and few null checks